### PR TITLE
Add homework (19.09.23)

### PR DESCRIPTION
**Pull request by abosiy**

### CODE REVIEW

File: home_work_7/src/com/bosiy/Book.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package com.bosiy;
+
+public class Book {
+    private String bookName;
+    private String writer;
+    private int yearOfPublishing;
+
+    public Book(String name) {
+        this.bookName = name;
+    }
```
Comment (scratchw0w) [inline home_work_7/src/com/bosiy/Book.java:10]: please add enter between methods

File: home_work_7/src/com/bosiy/Book.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    public void allInfoFirstAttribute() {
+        System.out.println("Name of book: " + bookName + "\n");
+    }
```
Comment (scratchw0w) [inline home_work_7/src/com/bosiy/Book.java]: and here

File: home_work_7/src/com/bosiy/Book.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    public Book(String name, String writer) {
+        this.bookName = name;
+        this.writer = writer;
+    }
```
Comment (scratchw0w) [inline home_work_7/src/com/bosiy/Book.java:15]: and here.. everywhere

File: home_work_7/src/com/bosiy/Book.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
+package com.bosiy;
+
+public class Book {
+    private String bookName;
```
Comment (scratchw0w) [inline home_work_7/src/com/bosiy/Book.java]: you should name it title in here

File: home_work_7/src/com/bosiy/Book.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    public Book(String name, String writer) {
+        this.bookName = name;
+        this.writer = writer;
+    }
+    public void allInfoSecondAttribute() {
+        System.out.println("Name of book: " + bookName + "\nWriter: " + writer + "\n");
+    }
+    public Book(String name, String writer, int yearOfPublishing) {
+        if(yearOfPublishing <= 0) {
```
Comment (scratchw0w) [inline home_work_7/src/com/bosiy/Book.java]: 0 is magic number, please move it to constants

also why aren't we able to create book with year of pub < 0? dont we have books which were written before

File: home_work_7/src/com/bosiy/Book.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        this.bookName = name;
+        this.writer = writer;
+        this.yearOfPublishing = yearOfPublishing;
+    }
+    public void allInfoThirdAttribute() {
+        System.out.println("Name of book: " + bookName + "\nWriter: " + writer +
+                "\nYear of publishing: " + yearOfPublishing + "\n");
+    }
+
+
```
Comment (scratchw0w) [inline home_work_7/src/com/bosiy/Book.java:22]: what's wrong with spaces here?

File: home_work_7/src/com/bosiy/Book.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        if(yearOfPublishing <= 0) {
+            System.out.println("The publication date of the book is incorrect!");
+            return;
+        }
+        this.bookName = name;
+        this.writer = writer;
+        this.yearOfPublishing = yearOfPublishing;
+    }
+    public void allInfoThirdAttribute() {
+        System.out.println("Name of book: " + bookName + "\nWriter: " + writer +
```
Comment (scratchw0w) [inline home_work_7/src/com/bosiy/Book.java]: use this.bookName etc.

File: home_work_7/src/com/bosiy/Book.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    public void setWriter(String writer) {
+        this.writer = writer;
+    }
+    public int getYearOfPublishing() {
+        return this.yearOfPublishing;
+    }
+
+    public void setYearOfPublishing(int yearOfPublishing) {
+        if(yearOfPublishing <= 0) {
```
Comment (scratchw0w) [inline home_work_7/src/com/bosiy/Book.java]: refer to upper comment

File: home_work_7/src/com/bosiy/MainApplication.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package com.bosiy;
+
+public class MainApplication {
+    public static void main(String[] args) {
+        Book firstBook = new Book("The Shinning");
```
Comment (scratchw0w) [inline home_work_7/src/com/bosiy/MainApplication.java]: consider different namings for each book

**Review by scratchw0w — CHANGES_REQUESTED** [review]

File: home_work_7/src/com/bosiy/Book.java (hunk 1 above)
Comment (abosiy) [reply home_work_7/src/com/bosiy/Book.java:10]: fixed 💪🏻

File: home_work_7/src/com/bosiy/Book.java (hunk 2 above)
Comment (abosiy) [reply home_work_7/src/com/bosiy/Book.java]: fixed

File: home_work_7/src/com/bosiy/Book.java (hunk 3 above)
Comment (abosiy) [reply home_work_7/src/com/bosiy/Book.java:15]: fixed

File: home_work_7/src/com/bosiy/Book.java (hunk 4 above)
Comment (abosiy) [reply home_work_7/src/com/bosiy/Book.java]: fixed

File: home_work_7/src/com/bosiy/Book.java (hunk 5 above)
Comment (abosiy) [reply home_work_7/src/com/bosiy/Book.java]: fixed

File: home_work_7/src/com/bosiy/Book.java (hunk 6 above)
Comment (abosiy) [reply home_work_7/src/com/bosiy/Book.java:22]: fixed

File: home_work_7/src/com/bosiy/Book.java (hunk 7 above)
Comment (abosiy) [reply home_work_7/src/com/bosiy/Book.java]: fixed

File: home_work_7/src/com/bosiy/Book.java (hunk 8 above)
Comment (abosiy) [reply home_work_7/src/com/bosiy/Book.java]: fixed

File: home_work_7/src/com/bosiy/MainApplication.java (hunk 9 above)
Comment (abosiy) [reply home_work_7/src/com/bosiy/MainApplication.java]: fixed

**Review by scratchw0w — APPROVED** [review]
Good Job!

**Pull request merged**